org.qt_project.qtcreator.yaml

```diff
@@ -0,0 +1,91 @@
+app-id: org.qt_project.qtcreator
+runtime: org.kde.Sdk
+runtime-version: "6.2"
+sdk: org.kde.Sdk
+command: qtcreator
+# Qt Creator already provides desktop launchers and AppData metadata
+# but since we cannot use the original AppID, we have to rename the original files
+rename-desktop-file: org.qt-project.qtcreator.desktop
+rename-appdata-file: org.qt-project.qtcreator.appdata.xml
+rename-icon: QtProject-qtcreator
+finish-args:
+# Qt Creator crashes GNOME Shell 3.36 on Wayland
+# - --socket=wayland
+# - --socket=fallback-x11
+- --socket=x11
+- --share=ipc
+- --share=network
+- --device=dri
+- --filesystem=host
+cleanup:
+- /include
+- /lib/arch
+- /lib/cmake
+- /lib/libQt6Core5Compat.prl
+- /lib/metatypes
+- /lib/qtcreator/objects-RelWithDebInfo
+- /lib/x86_64-linux-gnu
+- /mkspecs
+- /modules
+- /qml
+- /share/cmake
+- /share/pkgconfig
+- "*.a"
+cleanup-commands:
+- mv /app/lib/*-linux-gnu/* /app/lib/
+
+modules:
+# addition Qt modules that are not part of the KDE SDK
+- name: qt5compat
+  sources:
+  - type: git
+    url: git://code.qt.io/qt/qt5compat.git
+    tag: v6.2.1
+  buildsystem: cmake-ninja
+  builddir: true
+  post-install:
+  # move cmake files into a known path because we cannot determine the arch triplet
+  # (i.e. "x86_64-linux-gnu") using shell commands or environment variables
+  - mkdir /app/lib/arch/
+  - mv /app/lib/*-linux-gnu/cmake /app/lib/arch/
+
+- name: QtCreator
+  sources:
+  - type: git
+    url: git://code.qt.io/qt-creator/qt-creator.git
+    tag: v6.0.1
+  buildsystem: simple
```

Comment:
You should be able to avoid the simple buildsystem. My pQt6 WebEngine PR](https://github.com/flathub/io.qt.qtwebengine.BaseApp/pull/29) might give you some hints.  
Specifically, look at how I fixed the includes dirs in the `pri` files from `mkspecs` of Qt modules installed in `/app`.

Reply:
Are you referring to the `qt6-qtwebengine` module in your PR's [io.qt.qtwebengine.BaseApp.json](https://github.com/tinywrkb/io.qt.qtwebengine.BaseApp/blob/rt62/io.qt.qtwebengine.BaseApp.json)? I cannot relate this to the problem that I have with the "Devel" component in the install step. The reason for using the `simple` build system is that I cannot control the installed components with `cmake-ninja` since I have to install the default component for the actual IDE and the Devel component for the plugin development. Using the same `post-install` command when with `cmake-ninja` does fail for unknown reasons. I already discussed this in https://github.com/flathub/flathub/pull/2712#discussion_r776100057.